tests/dragonfly/conftest.py

```diff
@@ -257,14 +257,54 @@ def gen_tls_cert(df_factory: DflyInstanceFactory):
     tls_server_cert = dfly_path + tls_server_cert_file_name
     step3 = fr'openssl x509 -req -in {tls_server_req} -days 1 -CA {ca_cert} -CAkey {ca_key} -CAcreateserial -out {tls_server_cert}'
     subprocess.run(step3, shell=True)
-    return tls_server_key_file_name, tls_server_cert_file_name
+
+    # Step 4
+    # Generate client certificates
+    tls_client_key_file_name = "client-key.pem"
+    tls_client_key = dfly_path + tls_client_key_file_name
+
+    tls_client_req = dfly_path + "client-req.pem"
+    step4 = rf'openssl req -newkey rsa:4096 -nodes -keyout {tls_client_key} -out {tls_client_req} -subj "/C=It/ST=RM/L=Rome/O=KK/OU=Comp/CN=Gr/emailAddress=random@gmail.com"'
+    subprocess.run(step4, shell=True)
+
+    tls_client_cert_file_name = "client-cert.pem"
+    tls_client_cert = dfly_path + tls_client_cert_file_name
+    step5 = fr'openssl x509 -req -in {tls_client_req} -days 1 -CA {ca_cert} -CAkey {ca_key} -CAcreateserial -out {tls_client_cert}'
+
+    subprocess.run(step5, shell=True)
+
+    return [tls_server_key, tls_server_cert, ca_cert, tls_client_key, tls_client_cert]
```

Comment:
I find the current design a bit confusing
I suggest creating a fixture called gen_ca_cert which will return {"ssl_ca_certs", <path>}
create a function gen_tls_cert (not a fixture)
create a fixture called gen_server_cert which will return {"tls_key_file", <path> , "tls_cert_file", <path>}
and a fixture called gen_client_cert which will return {"ssl_keyfile", <path>, "ssl_certfile" , path}

In the test where you need to pass no_tls_on_admin_port , this will be done inside the test and not from the fixture
I think with the suggested design it will be more clear which args are passed to each test case